File: frontend/src/types/listening-record.ts
export interface ListeningRecord {
  id: string;
  user_id: string;
  episode_id: string;
  created_at: string;
}

export interface ListeningStatus {
  listened: boolean;
  listened_at?: string;
}

export interface ListeningRecordEpisode {
  id: string;
  title: string;
  podcast_id: string;
  artwork_url: string | null;
  published_at: string | null;
}

export interface ListeningRecordPodcast {
  id: string;
  title: string;
  artwork_url: string | null;
}

export interface ListeningRecordItem {
  id: string;
  episode: ListeningRecordEpisode;
  podcast: ListeningRecordPodcast;
  created_at: string;
}

export interface ListeningRecordListResult {
  records: ListeningRecordItem[];
  total: number;
}

File: backend/internal/model/listening_record.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// ListeningRecord はユーザーの聴取記録を表すドメインモデルです。
// ユーザーがエピソードを「聴いた」ことを記録します。
type ListeningRecord struct {
	ID        uuid.UUID `db:"id" json:"id"`
	UserID    uuid.UUID `db:"user_id" json:"user_id"`
	EpisodeID uuid.UUID `db:"episode_id" json:"episode_id"`
	CreatedAt time.Time `db:"created_at" json:"created_at"`
}
